.github/workflows/tag_on_version_update.yml

```diff
@@ -0,0 +1,47 @@
+name: Tag
+
+on:
+    pull_request:
+        branches:
+            - main
+        paths:
+            - "internal/version/version.go"
+        types:
+            - closed
+    push:
+
+jobs:
+    new_tag:
+        # if: github.event.pull_request.merged == true
```

Comment:
I guess this should be un-commented if we want to make sure it is triggered on merged PRs.